Tests/DatadogTests/Datadog/DatadogCore/KronosClockPublisherTests.swift

```diff
@@ -0,0 +1,67 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+@testable import Datadog
+
+class KronosClockPublisherTests: XCTestCase {
+    func testPickRandomDatadogNTPServers() throws {
+        let kronos = KronosClockMock()
+        let publisher = KronosClockPublisher(kronos: kronos)
+
+        var pools: Set<String> = []
+
+        try (0..<100).forEach { _ in
+            publisher.publish { _ in }
+            let pool = try XCTUnwrap(kronos.currentPool)
+            XCTAssertTrue(pool.hasSuffix(".datadog.pool.ntp.org"))
+            pools.insert(pool)
+        }
+
+        XCTAssertEqual(pools, Set(DatadogNTPServers), "Each time Datadog NTP server should be picked randomly.")
+    }
+
```

Comment:
We're missing coverage for one more logic branch in `KronosClock` - receiving an update on first calculated time value. In `KronosClockProtocol`, this corresponds to `first` parameter:
```swift
    /// - parameter first:      A closure that will be called after the first valid date is calculated.
    /// - parameter completion: A closure that will be called after _all_ the NTP calls are finished.
```

Reply:
Indeed, good catch!